minetest/minetest
# Add BMP test nodes

## Goal of the PR

- Test BMP rendering
- Test BMP deprecation detection

## How does the PR work?

This patch adds two test nodes for the BMP texture format:

- a 24bpp test node that loads a 2×2 RGB bitmap from a file
- a 24bpp test node that loads a base64-encoded in-band 2×2 RGB bitmap

## Does it resolve any reported issue?

This patch is adjacent to this issue: https://github.com/minetest/minetest/issues/13918
It tests the BMP deprecation feature from this PR: https://github.com/minetest/minetest/pull/13922

The test nodes are intended to check that BMP deprecation detection works.
They also have a hint in the description that BMP textures are deprecated (just to make it very clear).

## Does this relate to a goal in [the roadmap](https://github.com/minetest/minetest/blob/master/doc/direction.md)?

Correctness:

- BMP rendering is tested (for 24bpp RGB only)
- BMP deprecation warnings are tested (for both loading a BMP from file and inline base64-encoded BMPs).

## If not a bug fix, why is this PR needed? What usecases does it solve?

It provides 2 test cases for the code in this PR: https://github.com/minetest/minetest/pull/13922

Even though the code in that PR seems trivial, the tests are also trivial.

Also I was the only person asking for tests and so it is just fair that I created and added them.

## How to test

1. Start Minetest with the commit from this PR.
2. Start a devtest game.
3. Right click while having the ”Bag of Everything” selected.
4. Search for “BMP” in the formspec.
5. Acquire and place both test nodes.
6. Count the number of BMP deprecation warnings Minetest outputs. I think that number should be 2 (but I am not sure).
7. Verify that the nodes look a bit like the screenshot below (ignore the colored artifacts at the edges please). Each test node should have a 2×2 texture on all sides with the colors being red, green, yellow, blue (going clockwise from top left pixel).

![minetest-580-dev-bmp-testnodes](https://github.com/minetest/minetest/assets/60905/96e7899f-2944-4817-a388-b40e08eec140)
